src/core/algorithms/nar/des/des.cpp

```diff
@@ -0,0 +1,89 @@
+#include "des.h"
+
+#include "algorithms/nar/value_range.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "model/types/types.h"
+
+namespace algos::des {
+using model::ValueRange;
+
+DES::DES() : NARAlgorithm({}) {
+    using namespace config::names;
+    RegisterOptions();
+}
+
+void DES::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+
+    DifferentialStrategy default_strategy = DifferentialStrategy::rand1Bin;
+    RegisterOption(Option{&population_size_, kPopulationSize, kDPopulationSize, 100});
+    RegisterOption(
+            Option{&num_evaluations_, kMaxFitnessEvaluations, kDMaxFitnessEvaluations, 1000});
+    RegisterOption(Option{&differential_options_.differential_scale, kDifferentialScale,
+                          kDDifferentialScale, 0.5});
+    RegisterOption(Option{&differential_options_.crossover_probability, kCrossoverProbability,
+                          kDCrossoverProbability, 0.9});
+    RegisterOption(Option{&differential_options_.differential_strategy, kDifferentialStrategy,
+                          kDDifferentialStrategy, default_strategy});
+}
+
+void DES::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kPopulationSize, kMaxFitnessEvaluations, kDifferentialScale,
+                          kCrossoverProbability, kDifferentialStrategy});
+}
+
+FeatureDomains DES::FindFeatureDomains(TypedRelation const* typed_relation) {
+    auto feature_domains = std::vector<std::shared_ptr<ValueRange>>();
```

Comment:
FeatureDomains are of const type (domain of any given feature doesn't change during execution) so it can't be done like this, but i can do `std::vector<std::shared_ptr<ValueRange>> feature_domains`